src/solc_0.8/defi/StakeTokenWrapper.sol

```diff
@@ -13,16 +13,19 @@ abstract contract StakeTokenWrapper is Context {
     mapping(address => uint256) internal _balances;
 
     constructor(IERC20 stakeToken) {
+        require(address(stakeToken) != address(0), "StakeTokenWrapper: zero address");
         _stakeToken = stakeToken;
     }
 
     function _stake(uint256 amount) internal virtual {
+        require(amount > 0, "StakeTokenWrapper: amount > 0");
         _totalSupply = _totalSupply + amount;
         _balances[_msgSender()] = _balances[_msgSender()] + amount;
         _stakeToken.safeTransferFrom(_msgSender(), address(this), amount);
     }
 
     function _withdraw(uint256 amount) internal virtual {
+        require(amount > 0, "StakeTokenWrapper: amount > 0");
         _totalSupply = _totalSupply - amount;
         _balances[_msgSender()] = _balances[_msgSender()] - amount;
         _stakeToken.safeTransfer(_msgSender(), amount);
```

Comment:
- missing setTrustedForwarder checks
- missing unit tests for all the requires

Reply:
- missing setTrustedForwarder checks -> done
- missing setContributionRules setRewardCalculator checks -> done
- missing unit tests for all the requires -> done
- missing setERC721MultiplierList checks -> not done -> there are a lot of different scenários for the arguments: 
    - balanceOf = true | no multipliers | ids
    - balanceOf = true | no multipliers | no ids -> log calculation
    - balanceOf = false | multipliers and ids. 

https://github.com/thesandboxgame/sandbox-smart-contracts/pull/660/commits/b4d2f1ac5c98a07005cb4d1b1fb35635311555f7